hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java

```diff
@@ -611,7 +630,12 @@ public synchronized CompletableFuture<Void> add(OMClientResponse response,
   private synchronized boolean canFlush() {
     try {
       while (currentBuffer.size() == 0) {
-        wait(Long.MAX_VALUE);
+        wait(1000L);
+        if  (currentBuffer.size() == 0) {
+          // Both buffers are empty, so notify twice
+          flushNotifier.notifyFlush();
+          flushNotifier.notifyFlush();
+        }
```

Comment:
How is this snippet confirming both buffers are empty? Would you need to check `readyBuffer` size as well?

Reply:
https://github.com/apache/ozone/pull/4621#discussion_r1181914364

Reply:
Make sense.

In this case we could just remove inner `if` right? So the while-loop would become:

```java
      while (currentBuffer.size() == 0) {
        // Both buffers are empty at this point, notify twice
        flushNotifier.notifyFlush();
        flushNotifier.notifyFlush();

        wait(1000L);
      }
```

Also now that we both asked the same question. It might be worth writing that down in the code comment. :)